scripts/apps/archive/directives/MediaMetadata.ts

```diff
@@ -17,7 +17,9 @@ export function MediaMetadata(userList, archiveService, metadata) {
             scope.label = (id) => id;
 
             getLabelNameResolver().then((_getLabelForFieldId) => {
-                scope.label = _getLabelForFieldId;
+                $timeout(() => {
```

Comment:
it would be probably better to use `scope.$applyAsync` to avoid triggering digest on the whole app

Reply:
@petrjasek could you elaborate on `$apply` vs `$applyAsync`? If I understand correctly, `applyAsync` "debounces" the function call, but works the same, no? Can you choose to trigger digest to the whole app vs a section?

Reply:
`$apply` only works if you trigger it outside angular digest, like in some jquery on click handler. `$applyAsync` will work also if it's called during digest, scheduling the call to next iteration. both can run on the directive scope only, so both are fine. problem is `$timeout` which triggers global digest

Reply:
I see. I didn't know that `$timeout` triggers a global digest. I was chatting with @GyanP and he said it worked to call `scope.$apply` after setting `loading` to false. Timeout won't be needed then.